### ISSUE TITLE
ImportError for SHAP GPUTree: DLL load failed while importing _cext_gpu

### ISSUE THREAD

**Issue by tobyych** [shap/shap#1932]
Hi there, I was looking into GPUTreeExplainer to take advantage of GPU to speed up shap values calculations for large data set. 

I tried to build the shap package similar to the link below:
https://github.com/slundberg/shap/issues/1650#issuecomment-746985816

I was using pyenv-win and has python 3.8.7 installed.

In the stack trace, it read that a DLL could not be found
```
File "...\tmp\shap\shap\explainers\_gpu_tree.py", line 105, in shap_values
    assert_import("cext_gpu")
  File "...\tmp\shap\shap\utils\_general.py", line 23, in assert_import
    raise e
  File "...\tmp\shap\shap\explainers\_gpu_tree.py", line 6, in <module>
    from .. import _cext_gpu
ImportError: DLL load failed while importing _cext_gpu: The specified module could not be found.
```

I then used [dependency walker](https://github.com/lucasg/Dependencies) in Windows to trace which DLL is missing, and found that `python38.dll` went missing for `_cext_gpu.cp38-win_amd64.pyd`

I tried to re-install Python 3.8.7 with Windows Installer from python.org and obtained the python38.dll, but I wasn't sure where to put it for the import to work. Highly appreciate if someone could point me to the correct way of doing this. Thanks!

**Comment by tobyych** [comment]
It turned out that `cudart64_110.dll` was missing. After adding the .dll file to the package directory, the exception was gone.
![image](https://user-images.githubusercontent.com/44737479/114498370-8ce6f980-9c56-11eb-9840-29d0ed25a954.png)

The output from dependency walker as mentioned above seemed to be incorrect. Using `dumpobj` that came with MSVC, I found that there were more dependent binaries. By putting all of these to the package directoy, I was able to run shap using GPU. And then I removed these binaries one by one and found only `cudart64_110.dll` mattered.
![image](https://user-images.githubusercontent.com/44737479/114498354-85bfeb80-9c56-11eb-9895-f5ef55b30bc6.png)

It felt suspicious since the directory where I copied `cudart64_110.dll` from was in the system PATH environment variable as shown below:
![image](https://user-images.githubusercontent.com/44737479/114498609-f9fa8f00-9c56-11eb-872d-b0b2f758b5ec.png)

In file explorer:
![image](https://user-images.githubusercontent.com/44737479/114498634-07177e00-9c57-11eb-9a91-552569727a0e.png)

However, the program was not able to search that file via the PATH variable. Does anyone know if that's an issue with my directories?
